cmake/extensions.cmake

```diff
@@ -1681,3 +1681,65 @@ function(generate_unique_target_name_from_filename filename target_name)
 
   set(${target_name} gen_${x}_${unique_chars} PARENT_SCOPE)
 endfunction()
+
+# Usage:
+#   zephyr_file(<mode> <arg> ...)
+#
+# Zephyr file function extension.
+# This function currently support the following <modes>
```

Comment:
A wise man asked me once why an optional argument that was always provided was necessary.

Similarly, if there's only one mode supported, does this really need this generality?

Reply:
> A wise man asked me once why an optional argument that was always provided was necessary.

correct, but the reason here is to follow CMake file function: https://cmake.org/cmake/help/latest/command/file.html#file

> Similarly, if there's only one mode supported, does this really need this generality?

In order to minimize the risk of someone needing an funtion that can be covered under `zephyr_file()` then I decided to keep the arg as an option and use `cmake_parse_arguments`.

That way, the next person will hopefully extend this function and not create one more.
An example of such extension could be:
`zephyr_file(ZEPHYR_BASE <path>)` which would do similar to `APPLICATION_ROOT` but using `ZEPHYR_BASE` instead.

Reply:
> the reason here is to follow CMake file function: https://cmake.org/cmake/help/latest/command/file.html#file

Seems like it's not really necessary unless you've got plans for extending this IMO, but I understand your rationale.

Reply:
>  unless you've got plans 

i'm not able to follow any plan that spans more than a week.
Everything that goes beyond a week is to be considered a loose idea :wink: